route rule: Sanitize IP network

cathay4t opened the pull request:
Supporting these IP format for route rule `ip-from` and `ip-to`:
* `192.0.2.1/24` == `192.0.2.0/24`
* `192.0.2.1` == `192.0.2.1/32`
* `2001:db8:1::1/64` == `2001:db8:1::0/64`
* `2001:db8:1::1` = `2001:db8:1::1/128`
* `2001:db8:1:0000:000::1` == `2001:db8:1::1/128`

Unit test cases and integration test case included.

ffmancera reviewed (changes requested):
`test_add_rules_to_new_interface` unit test is failing, could you take a look? Thanks

cathay4t committed:
route rule: Sanitize IP network

Supporting these IP format for route rule `ip-from` and `ip-to`:
 * `192.0.2.1/24` == `192.0.2.0/24`
 * `192.0.2.1` == `192.0.2.1/32`
 * `2001:db8:1::1/64` == `2001:db8:1::0/64`
 * `2001:db8:1::1` = `2001:db8:1::1/128`
 * `2001:db8:1:0000:000::1` == `2001:db8:1::1/128`

Unit test cases and integration test case included.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
Unit test case fixed. Thanks!

cathay4t committed:
test: Retry on assertion of test_desire_port_only_should_preserve_ctrl_setting

In github CI system, we see a rare failure on
`test_desire_port_only_should_preserve_ctrl_setting` showing bridge lose
the port. It might caused by slow CI system where NetworkManager is
still attaching port to bridge.

We retry when asserting the bridge port list with 5 seconds timeout.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
In github CI system, we see a rare failure on
`test_desire_port_only_should_preserve_ctrl_setting` showing bridge lose
the port. It might caused by slow CI system where NetworkManager is
still attaching port to bridge.

Included new pathc to retry when asserting the bridge port list with 5 seconds timeout.

For the CI failure of slow system, it most likely been caused by heavy loaded github CI system.
